source/architecture.rst

```diff
@@ -1,230 +1,245 @@
+a bottleneck for the architecture.
 
-To keep a certain level of compatibility with Fiware services, meaning, for using Fiware services and components in the
-*dojot* platform with the minimal amount of adaptations, we added a NGSI interface translation service.
+To keep a certain level of compatibility with NGSI-compatible components, we
+added a NGSI interface translation service.
 
-Device Manager
+DeviceManager
 **************
 
-The IoT Device manager is a core entity, responsible for maintaining the devices data models and its abstractions, it is
-also responsible for propagating this models to services that are interested in this kind of information, for example,
-the iot-agent.
+DeviceManager is a core entity which is responsible for keeping device and
+templates data models. It is also responsible for publishing any updates to all
+interested components (namely IoT agents, history and subscription manager)
+through Kafka.
 
-This service is stateless, having its data persisted to a database, with data isolation for users and applications,
-making possible a multi-tenant architecture for the middleware.
+This service is stateless, having its data persisted to a database, with data
+isolation for users and applications, making possible a multi-tenant
+architecture for the middleware.
 
-Iot-agent
+IoT Agent
 *********
 
-The iot-agent is an adaptation service between the data model of the context broker and the devices data models. The
-*dojot* platform can have multiple iot-agents, each one of them being specialized in a specific protocol like, for
-instance, MQTT/JSON, CoAP/LWM2M and HTTP/JSON.
+An IoT agent is an adaptation service between physical devices and *dojot* core
+components. It could be understood as a *device driver* for a set of devices.
+The *dojot* platform can have multiple iot-agents, each one of them being
+specialized in a specific protocol like, for instance, MQTT/JSON, CoAP/LWM2M
+and HTTP/JSON.
+
+It is also responsible for all security mechanisms used in communication
+channels between the devices it deals with and the component itself.
 
-Security measures like the management of the secure channel used for the communication of the platform with the device
-is also treated by this service.
 
 User Authorization Service
 **************************
 
-This service is responsible for managing user profiles and access control. Basically any API call that reaches the
-platform via the API Gateway is validated by this service.
+This service is responsible for managing user profiles and access control.
+Basically any API call that reaches the platform via the API Gateway is
+validated by this service.
 
-To be able to deal with a high volume of authorization calls, it uses caching, it is stateless and it is scalable
-horizontally. Its data is stored on a database.
+To be able to deal with a high volume of authorization calls, it uses caching,
+it is stateless and it is scalable horizontally. Its data is stored on a
```

Comment:
using cache makes it stateful, doesn't it?

Reply:
I don't think so - because caching things won't change the answer based on a cached (or non-cached) information. It won't save the last requested device so that future requests will have different answers because of this.

Reply:
ok